File: platform/sdk/python/examples/graph/graph-011_spark_idempotent_upsert.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from mox_sdk_graph import GraphClient

client = GraphClient({"endpoint": "graph.local"})
nodes = [
    {"id": "nu1", "label": "User"},
    {"id": "nu2", "label": "User"},
    {"id": "nu1", "label": "User"}
]
result = client.sparkIdempotentUpsert(nodes)
if not result["idempotent"] or result["updated"] < 1:
    sys.exit(1)
print("XJ-OK: graph-011_spark_idempotent_upsert")
sys.exit(0)

File: platform/sdk/nodejs/examples/graph/graph-011_spark_idempotent_upsert.js
const { GraphClient } = require("../../mox-sdk-graph");
const client = new GraphClient({ endpoint: "graph.local" });
const nodes = [
  { id: "nu1", label: "User" },
  { id: "nu2", label: "User" },
  { id: "nu1", label: "User" }
];
const result = client.sparkIdempotentUpsert(nodes);
if (!result.idempotent || result.updated < 1) process.exit(1);
console.log("XJ-OK: graph-011_spark_idempotent_upsert");
process.exit(0);
